File: app/src/main/java/org/odddev/fantlab/award/AwardsFragment.java
package org.odddev.fantlab.award;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.design.widget.Snackbar;
import android.support.v4.view.GravityCompat;
import android.support.v4.widget.DrawerLayout;
import android.support.v7.app.ActionBar;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.LinearLayoutManager;
import android.view.LayoutInflater;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;

import com.arellomobile.mvp.MvpAppCompatFragment;
import com.arellomobile.mvp.presenter.InjectPresenter;

import org.odddev.fantlab.R;
import org.odddev.fantlab.databinding.AwardsFragmentBinding;

import java.util.List;

/**
 * @author kenrube
 * @since 10.12.16
 */

public class AwardsFragment extends MvpAppCompatFragment implements IAwardsView {

	private AwardsFragmentBinding binding;
	private AwardsAdapter adapter;

	@InjectPresenter
	AwardsPresenter presenter;

	@Nullable
	@Override
	public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
							 @Nullable Bundle savedInstanceState) {
		binding = AwardsFragmentBinding.inflate(inflater, container, false);
		return binding.getRoot();
	}

	@Override
	public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
		presenter.getAwards();

		initToolbar();
		setHasOptionsMenu(true);

		initRecyclerView();
	}

	private void initToolbar() {
		AppCompatActivity activity = (AppCompatActivity) getActivity();
		activity.setSupportActionBar(binding.toolbar);
		ActionBar actionBar = activity.getSupportActionBar();
		if (actionBar != null) {
			actionBar.setTitle(R.string.nav_awards);
			actionBar.setHomeAsUpIndicator(R.drawable.ic_menu_white_24dp);
			actionBar.setDisplayHomeAsUpEnabled(true);
		}
	}

	private void initRecyclerView() {
		LinearLayoutManager layoutManager = new LinearLayoutManager(getContext());
		binding.awards.setLayoutManager(layoutManager);
		adapter = new AwardsAdapter();
		binding.awards.setAdapter(adapter);
	}

	@Override
	public boolean onOptionsItemSelected(MenuItem item) {
		switch (item.getItemId()) {
			case android.R.id.home: {
				((DrawerLayout) getActivity().findViewById(R.id.drawer_layout))
						.openDrawer(GravityCompat.START);
				break;
			}
		}
		return super.onOptionsItemSelected(item);
	}

	@Override
	public void showAwards(List<Award> awards) {
		adapter.setAwards(awards);
	}

	@Override
	public void showError(String message) {
		Snackbar.make(binding.getRoot(), message, Snackbar.LENGTH_LONG).show();
	}
}

File: app/src/main/java/org/odddev/fantlab/award/AwardsFragment.kt
package org.odddev.fantlab.award

import android.os.Bundle
import android.support.design.widget.Snackbar
import android.support.v4.view.GravityCompat
import android.support.v4.widget.DrawerLayout
import android.support.v7.app.AlertDialog
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.LinearLayoutManager
import android.view.*

import com.arellomobile.mvp.MvpAppCompatFragment
import com.arellomobile.mvp.presenter.InjectPresenter

import org.odddev.fantlab.R
import org.odddev.fantlab.databinding.AwardsFragmentBinding

/**
 * @author kenrube
 * *
 * @since 10.12.16
 */

class AwardsFragment : MvpAppCompatFragment(), IAwardsView {

	private lateinit var binding: AwardsFragmentBinding
	private lateinit var adapter: AwardsAdapter

	@InjectPresenter
	lateinit var presenter: AwardsPresenter

	override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?,
							  savedInstanceState: Bundle?): View? {
		binding = AwardsFragmentBinding.inflate(inflater, container, false)
		return binding.root
	}

	override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
		presenter.getAwards()

		initToolbar()
		setHasOptionsMenu(true)

		initRecyclerView()
	}

	private fun initToolbar() {
		val activity = activity as AppCompatActivity
		activity.setSupportActionBar(binding.toolbar)
		val actionBar = activity.supportActionBar
		if (actionBar != null) {
			actionBar.setTitle(R.string.nav_awards)
			actionBar.setHomeAsUpIndicator(R.drawable.ic_menu_white_24dp)
			actionBar.setDisplayHomeAsUpEnabled(true)
		}
	}

	private fun initRecyclerView() {
		val layoutManager = LinearLayoutManager(context)
		binding.awards.layoutManager = layoutManager
		adapter = AwardsAdapter()
		binding.awards.adapter = adapter
	}

	override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
		inflater.inflate(R.menu.action_bar_awards, menu)
	}

	override fun onOptionsItemSelected(item: MenuItem): Boolean {
		when (item.itemId) {
			android.R.id.home -> (activity.findViewById(R.id.drawer_layout) as DrawerLayout)
					.openDrawer(GravityCompat.START)
			R.id.filter_by -> presenter.onFilterButtonClick()
		}
		return super.onOptionsItemSelected(item)
	}

	override fun showFilterDialog() {
		AlertDialog.Builder(context)
				.setTitle(getString(R.string.award_filter_dialog_title))
				.setView(R.layout.award_filter_dialog)
				.setPositiveButton(android.R.string.ok, null)
				.setNegativeButton(getString(R.string.award_filter_cancel), null)
				.show()
	}

	override fun showAwards(awards: List<Award>) {
		adapter.setAwards(awards)
	}

	override fun showError(message: String) {
		Snackbar.make(binding.root, message, Snackbar.LENGTH_LONG).show()
	}
}
